crates/egui_plot/src/transform.rs

```diff
@@ -117,14 +117,22 @@ impl PlotBounds {
 
     #[inline]
     pub fn expand_x(&mut self, pad: f64) {
-        self.min[0] -= pad;
-        self.max[0] += pad;
+        if (self.min[0] - pad).is_finite() {
+            self.min[0] -= pad;
+        }
+        if (self.max[0] + pad).is_finite() {
+            self.max[0] += pad;
+        }
```

Comment:
Alternatively:

```
self.min[0] = (self.min[0] - pad).clamp(f64::MIN, f64::MAX);
self.max[0] = (self.max[0] + pad).clamp(f64::MIN, f64::MAX);
```

or add a `self.clamp_to_finite();` helper to call at the end of these functions that does


```
fn clamp_to_finite(&mut self) {
        for d in 0..2 {
			self.min[d] = self.min[d].clamp(f64::MIN, f64::MAX);
			self.max[d] = self.max[d].clamp(f64::MIN, f64::MAX);
        }
}
```

Reply:
Hi,

I don't think so
The goal of these if is to ensure no propagation of `NaN`

The docs said if NaN is input of clamp, it will return NaN - a problem

Reply:
```rs
    pub fn expand_x(&mut self, pad: f64) {
        if pad.is_finite() {
		        self.min[0] -= pad;
		        self.max[0] += pad;
				self.clamp_to_finite();
		}
	}


	fn clamp_to_finite(&mut self) {
	        for d in 0..2 {
				self.min[d] = self.min[d].clamp(f64::MIN, f64::MAX);
	            if self.min[d].is_nan() { self.min[d] = 0.0; }
	
				self.max[d] = self.max[d].clamp(f64::MIN, f64::MAX);
	            if self.max[d].is_nan() { self.max[d] = 0.0; }
	        }
	}
```